friktion/friktion/bid_details.py

```diff
@@ -1,25 +1,28 @@
 from dataclasses import dataclass
+from typing import Tuple
 
 from anchorpy import Wallet
 from nacl import signing  # type: ignore
 from solana.publickey import PublicKey
-
+from solders.signature import Signature
 
 @dataclass
 class BidDetails:
     signer_wallet: PublicKey
     order_id: int
     bid_size: int
     bid_price: int
+    referrer: PublicKey
 
-    def as_signed_msg(self, wallet: Wallet) -> signing.SignedMessage:
+    def as_signed_msg(self, wallet: Wallet) -> Tuple[bytes, Signature]:
         give_amount = self.bid_size
         receive_amount = self.bid_price * self.bid_size
         byte_rep = b"".join(
             [
                 bytes([self.order_id]),
                 wallet.public_key.__bytes__(),
+                self.referrer.__bytes__(),
                 bytes([give_amount, receive_amount]),
             ]
         )
-        return wallet.payer.sign(byte_rep)
+        return (byte_rep, wallet.payer.sign(byte_rep))
```

Comment:
the return of `wallet.payer.sign(byte_rep)` is not of type `Signature`


![Screen Shot 2022-07-28 at 10 04 00](https://user-images.githubusercontent.com/4041238/181511939-47a8c2c5-a099-42ed-82a8-b330d310e257.png)

Reply:
Weird, that's what I'm seeing locally, and mypy isn't catching the error

Reply:
![image](https://user-images.githubusercontent.com/89875647/181651769-5614d220-6614-4d9f-b6b8-e5364d8db592.png)

might have been fixed in my newest updates?

Reply:
Maybe you are missing to push some changes?
I checkout out to the newest commit `3e978027bcf8` but I am still getting `SignedMessage` as the return of `as_signed_msg`

![Screen Shot 2022-07-29 at 08 53 36](https://user-images.githubusercontent.com/4041238/181753148-23724228-aab8-4543-881b-57e3a7ea0196.png)
![Screen Shot 2022-07-29 at 08 56 42](https://user-images.githubusercontent.com/4041238/181753618-42f20438-9d7f-4584-90e4-bd8111877061.png)

Reply:
My bad. Updating solana and anchorpy fixes the issue